### PR TITLE
misc: snapshotter: Add `k3s` and `rke2` overlays to the snapshotter daemonset

### PR DESCRIPTION
**Pull request by fidencio**
This is a small series that brings support to using the nydus-snapshotter daemonset with different flavours of kubernetes.

The background of why we're doing this, is mainly because we rely on those in the Kata Containers CI.

Please, take a look at each commit message for more details.

### CODE REVIEW

**Comment by fidencio** [comment]
cc @ChengyuZhu6

File: misc/snapshotter/snapshotter.sh [hunk 1]
```diff
@@ -227,13 +219,59 @@ function cleanup_snapshotter() {
...
+        if [ ! -f "${CONTAINER_RUNTIME_CONFIG}" ]; then
+            mkdir -p $(dirname ${CONTAINER_RUNTIME_CONFIG}) || true
+            if [ -x $(command -v ${CONTAINER_RUNTIME}) ]; then
+                ${CONTAINER_RUNTIME} config default > ${CONTAINER_RUNTIME_CONFIG}
+            else
+                die "Not able to find an executable ${CONTAINER_RUNTIME} binary to create the default config"
+            fi
+        fi
+    else
+        die "${CONTAINER_RUNTIME} is a unsupported containe rruntime"
```
Comment (imeoer) [inline misc/snapshotter/snapshotter.sh:272]: Typo `rruntime`.

Comment (fidencio) [reply misc/snapshotter/snapshotter.sh]: Fixed, thanks!

**Comment by ChengyuZhu6** [comment]
> @imeoer, I'm trying to understand whether the E2E failures are related to this PR or not.

@fidencio You can display the pod's log for e2e test.

**Comment by fidencio** [comment]
@imeoer, I was able to get the full Ci green, please, let me know if anything else is needed from my side.

**Review by ChengyuZhu6 — APPROVED** [review]

**Review by imeoer — APPROVED** [review]
LGTM!

**Pull request merged**